src/wallet/wallet.cpp

```diff
@@ -811,6 +812,23 @@ void CWallet::IncrementNoteWitnesses(const CBlockIndex* pindex,
                 }
             }
         }
+        // Sapling
+        for (uint32_t i = 0; i < tx.vShieldedOutput.size(); i++) {
+            const OutputDescription& outputDesc = tx.vShieldedOutput[i];
+            const uint256& note_commitment = outputDesc.cm;
+            saplingTree.append(note_commitment);
+
+            // Increment existing witnesses
+            for (std::pair<const uint256, CWalletTx>& wtxItem : mapWallet) {
+                ::AppendNoteCommitment(wtxItem.second.mapSaplingNoteData, pindex->nHeight, nWitnessCacheSize, note_commitment);
+            }
+
+            // If this is our note, witness it
```

Comment:
Nit: Now that `WitnessNote()` has been extracted, this comment doesn't make as much sense (because `txIsOurs` only says that *something* in the transaction is ours; we check if it is *this* note inside `WitnessNote()`). We should think about rewording it, or possibly renaming `WitnessNote()` to `WitnessNoteIfMine()`.

Reply:
I had also considered calling this method MaybeWitnessNote and passing in txIsOurs, but I didn't like that, and couldn't come up with something better in two minutes so I left it. If you have something you like I can change it, otherwise I am relatively happy leaving it as is.

Reply:
I actually just went ahead and made the name change.